client/gutenberg/editor/controller.js

```diff
@@ -3,14 +3,38 @@
  * External dependencies
  */
 import React from 'react';
+import { startsWith } from 'lodash';
 
 /**
  * Internal dependencies
  */
 import GutenbergEditor from 'gutenberg/editor/main';
 
+function getPostID( context ) {
+	if ( ! context.params.post || 'new' === context.params.post ) {
+		return null;
+	}
+
+	// both post and site are in the path
+	return parseInt( context.params.post, 10 );
+}
+
+function determinePostType( context ) {
+	if ( startsWith( context.path, '/gutenberg/post' ) ) {
+		return 'post';
+	}
+
+	if ( startsWith( context.path, '/gutenberg/page' ) ) {
+		return 'page';
+	}
+
+	return context.params.type;
+}
```

Comment:
can we not set these in the routing path?

```js
page( '/gutenberg/:type', … )
```

Reply:
Maybe we can verify if the route can be defined in such a way that it's passed as a context parameter instead. I do understand that sometimes we're too clever with out routing scheme (for example site slugs being appended to the end of just about any route ) that makes this not possible.

We should also consider CPTs as well, though for this exploration, this is a more minor point.